extensions/wikia/UserLogin/UserSignupSpecialController.class.php

```diff
@@ -122,82 +125,84 @@ public function signupForm() {
 		// process signup
 		$redirected = $this->request->getVal( 'redirected', '' );
 		if ( $this->wg->Request->wasPosted() && empty( $redirected ) ) {
+			$this->handleSignupFormSubmit();
+		} else {
+			$this->track( 'signup-start' );
```

Comment:
All good and valid points! I think the older versions of PHP, not using `this` or `static` for static methods was problematic or at least discouraged. Now it is not the case, of course.